### PR TITLE
Sdktechno 66

### PR DESCRIPTION
**Pull request by finalspy**
Remove python 2.7 for spark 3.0 due to deprecation.
Adapt entrypoint.sh to reflect changes for spark3.0.
Add pyspark files to env variable for python executors.

### CODE REVIEW

File: technologies/job/spark/spark-3.0/innerContexts/python/entrypoint.sh [hunk 1]
```diff
@@ -85,58 +65,58 @@ if [ -n "$PYSPARK_FILES" ]; then
...
+SPARK_CLASSPATH="$SPARK_CLASSPATH:${SPARK_HOME}/jars/*"
+env | grep SPARK_JAVA_OPT_ | sort -t_ -k4 -n | sed 's/[^=]*=\(.*\)/\1/g' > /tmp/java_opts.txt
+readarray -t SPARK_EXECUTOR_JAVA_OPTS < /tmp/java_opts.txt
 
+if [ -n "$SPARK_EXTRA_CLASSPATH" ]; then
+  SPARK_CLASSPATH="$SPARK_CLASSPATH:$SPARK_EXTRA_CLASSPATH"
+fi
 
-if [ "$PYSPARK_MAJOR_PYTHON_VERSION" == "3" ]; then
+if [ "$PYSPARK_MAJOR_PYTHON_VERSION" == "2" ]; then
```
Comment (finalspy) [inline technologies/job/spark/spark-3.0/innerContexts/python/entrypoint.sh:79]: Cette partie issue du code spark peut être virée amha

File: technologies/job/spark/spark-3.0/entrypoint.sh [hunk 2]
```diff
@@ -30,18 +30,25 @@ set -e
 # If there is no passwd entry for the container UID, attempt to create one
 if [ -z "$uidentry" ] ; then
     if [ -w /etc/passwd ] ; then
-        echo "$myuid:x:$myuid:$mygid:anonymous uid:$SPARK_HOME:/bin/false" >> /etc/passwd
+        echo "$myuid:x:$myuid:$mygid:${SPARK_USER_NAME:-anonymous uid}:$SPARK_HOME:/bin/false" >> /etc/passwd
```
Comment (PierreLeresteux) [inline technologies/job/spark/spark-3.0/entrypoint.sh:33]: Is it necessary ? (not added in the PR description)

Comment (finalspy) [reply technologies/job/spark/spark-3.0/entrypoint.sh:33]: It comes from spark code : https://github.com/apache/spark/blob/a7d0d353cd745391cf285039a4630b37754a56a0/resource-managers/kubernetes/docker/src/main/dockerfiles/spark/entrypoint.sh

File: technologies/job/spark/spark-3.0/innerContexts/python/entrypoint.sh (hunk 1 above)
Comment (finalspy) [reply technologies/job/spark/spark-3.0/innerContexts/python/entrypoint.sh:79]: FYI : https://github.com/apache/spark/blob/a7d0d353cd745391cf285039a4630b37754a56a0/resource-managers/kubernetes/docker/src/main/dockerfiles/spark/entrypoint.sh

**Review by PierreLeresteux — APPROVED** [review]

**Pull request merged**